.pfnci/trim_cupy_kernel_cache.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+"""
+A tool to clean-up least-recently-used CuPy kernel caches.
+
+The expiry is the length to invalidate unused CuPy cache, in seconds.
+For example, setting the expiry to 3,600 will trim all kernel caches
+not being used within an hour. The maximum number of files or maximum total
+size can also be specified to cap the amount of caches.
+
+Note that this code relies on atime support of the filesystem.
+"""
+
+import argparse
+import datetime
+import glob
+import os
+import sys
+
+
+def _log(msg):
+    sys.stderr.write(msg + '\n')
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--expiry', type=int, default=None)
+    parser.add_argument('--max-files', type=int, default=None)
+    parser.add_argument('--max-size', type=int, default=None)
+    parser.add_argument('--print', default=False, action='store_true')
+    parser.add_argument('--rm', default=False, action='store_true')
+    options = parser.parse_args()
+
+    cache_dir = os.environ.get(
+            'CUPY_CACHE_DIR',
+            os.path.expanduser('~/.cupy/kernel_cache/'))
```

Comment:
Is this path safe?

Reply:
Yes, it's the same as we do in cupy: https://github.com/cupy/cupy/blob/065adda84f751f69593540baf7eaccd0c5350e9d/cupy/cuda/compiler.py#L375-L379